src/tests/src/libs/antares/study/parallel-years/test-parallel-years-calculator.cpp

```diff
@@ -0,0 +1,354 @@
+                                         enableParallel,
+                                         maxNbYearsInParallel, 8,
+                                         thermalTSRefresh,
+                                         params);
+
+    BOOST_CHECK_EQUAL(builder.getForcedNbOfParallelYears(), 4);
+    BOOST_CHECK(builder.allSetsParallelYearsHaveSameSize());
+    BOOST_CHECK_EQUAL(builder.getMinNbParallelYearsForGUI(), 4);
+    BOOST_CHECK_EQUAL(builder.getNbYearsReallyPerformed(), 100);
+}
+
+BOOST_AUTO_TEST_CASE(hundred_mc_years_no_force_enable_parallel_thermal_refresh_on_draft_mode)
+{
+
+    bool thermalTSRefresh = true;
+    forceParallel = false;
+    enableParallel = true;
+
+    params.mode = Antares::Data::stdmAdequacyDraft;
+
+    params.nbYears = 100;
+
+    params.nbCores.ncMode = Antares::Data::NumberOfCoresMode::ncAvg;
+
+    SetsOfParallelYearCalculator builder(forceParallel,
+                                         enableParallel,
+                                         maxNbYearsInParallel, 3,
+                                         thermalTSRefresh,
+                                         params);
+
+    BOOST_CHECK_EQUAL(builder.getForcedNbOfParallelYears(), 1);
+    BOOST_CHECK(builder.allSetsParallelYearsHaveSameSize());
+    BOOST_CHECK_EQUAL(builder.getMinNbParallelYearsForGUI(), 1);
+    BOOST_CHECK_EQUAL(builder.getNbYearsReallyPerformed(), 100);
+}
+
+BOOST_AUTO_TEST_CASE(hundred_mc_years_force_parallel_enable_parallel_thermal_refresh_on_hot_start)
+{
+
+    bool thermalTSRefresh = true;
+    forceParallel = true;
+    enableParallel = true;
+
+    maxNbYearsInParallel = 5;
+
+    params.initialReservoirLevels.iniLevels = Antares::Data::irlHotStart;
+
+    params.nbYears = 100;
+
+    params.nbCores.ncMode = Antares::Data::NumberOfCoresMode::ncAvg;
+
+    SetsOfParallelYearCalculator builder(forceParallel,
+                                         enableParallel,
+                                         maxNbYearsInParallel, 4,
+                                         thermalTSRefresh,
+                                         params);
+
+    BOOST_CHECK_EQUAL(builder.getForcedNbOfParallelYears(), 5);
+    BOOST_CHECK(builder.allSetsParallelYearsHaveSameSize());
+    BOOST_CHECK_EQUAL(builder.getMinNbParallelYearsForGUI(), 5);
+    BOOST_CHECK_EQUAL(builder.getNbYearsReallyPerformed(), 100);
+}
+
+
+BOOST_AUTO_TEST_CASE(hundred_mc_years_force_10_parallel_user_playlist_thermal_refresh_on)
+{
+
+    bool thermalTSRefresh = true;
+    forceParallel = true;
+    enableParallel = true;
+    
+    maxNbYearsInParallel = 10;
+
+    params.nbYears = 100;
+
+    // Create playlist
+    params.userPlaylist = true;
+    params.yearsFilter.reserve(params.nbYears);
+    for (uint i = 0; i != params.nbYears; ++i)
+        params.yearsFilter[i] = false;
+    params.yearsFilter[2] = true;
+    params.resetYearsWeigth();
+    params.setYearWeight(0,4);
+    params.setYearWeight(1,10);
+    params.setYearWeight(2,3);
```

Comment:
Why do we need to set some years' weight to test the parallel years orchestrator (or builder) ?
It seems to me that year weights and  parallel years orchestrator are not related.